### PR TITLE
add e2e test for cacert rotation

### PR DESCRIPTION
**Pull request by zirain**
**Please provide a description of this PR:**

follow up: https://github.com/istio/istio/pull/47424

### CODE REVIEW

**Review by dhawton — APPROVED** [review]

File: tests/integration/security/cacert_rotation/main_test.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    defaultConfig:
+      proxyMetadata:
+        PROXY_CONFIG_XDS_AGENT: "true"
+`
+	cfg.ControlPlaneValues = cfgYaml
+}
+
+func TestReachability(t *testing.T) {
+	framework.NewTest(t).
+		Features("security.peer.multiple-root").
```
Comment (hzxuzhonghu) [inline tests/integration/security/cacert_rotation/main_test.go:79]: This is not right, i think this is from multi root test

File: tests/integration/security/cacert_rotation/main_test.go [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+				}
+
+				// retry when workload cert is not updated
+				if updateTime.After(lastUpdateTime) {
+					t.Logf("workload cert is updated, last update time: %v", updateTime)
+					return true
+				}
+
+				t.Logf("workload cert is not updated, last update time: %v", lastUpdateTime)
+				return true
```
Comment (hzxuzhonghu) [inline tests/integration/security/cacert_rotation/main_test.go:134]: should false?

File: tests/integration/security/cacert_rotation/main_test.go [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+					return false
+				}
+
+				// retry when workload cert is not updated
+				if updateTime.After(lastUpdateTime) {
+					t.Logf("workload cert is updated, last update time: %v", updateTime)
+					return true
+				}
+
+				t.Logf("workload cert is not updated, last update time: %v", lastUpdateTime)
```
Comment (hzxuzhonghu) [inline tests/integration/security/cacert_rotation/main_test.go:133]: debug

**Review by hzxuzhonghu — COMMENTED** [review]
can we test the case, one with old cert and the other new cert

**Comment by zirain** [comment]
> can we test the case, one with old cert and the other new cert

It's hard to find such a precise moment now.
Alternative: Keep sending traffic, all traffic result should be 200 all the time(before and after cert rotation).

File: tests/integration/security/cacert_rotation/main_test.go [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+				// retry when workload cert is not updated
+				if updateTime.After(lastUpdateTime) {
+					t.Logf("workload cert is updated, last update time: %v", updateTime)
+					return true
+				}
+
+				return false
+			}, retry.Timeout(time.Minute), retry.Delay(1*time.Second))
+
+			// Verify traffic works between a and b after cert rotation
```
Comment (hzxuzhonghu) [inline tests/integration/security/cacert_rotation/main_test.go:136]: Maybe it is feasible to startup a go thread running calling during the waiting, just not sure how to control the calling numbers

**Review by hzxuzhonghu — APPROVED** [review]

**Review by ericvn — APPROVED** [review]

**Comment by jaellio** [comment]
@zirain If #47424 was backported to v1.20 then the e2es should also be backported.

**Pull request merged**